[Docs] Use Pure Dataclass In Example (#5829)
flyteorg/flyte: docs/user_guide/data_types_and_io/dataclass.md

to serialize and deserialize dataclasses.

:::{important}

> If you're using Flytekit version below v1.11.1, your dataclass will need to decorate with `@dataclass_json` or

Reviewer: ```suggestion
If you're using Flytekit version greater than v1.10 and less than v1.11.1, you will need add `from mashumaro.mixins.json import DataClassJSONMixin` to your imports and either decorate your dataclass with `@dataclass_json` or create a class that inherits from `DataClassJSONMixin`. 
```

Reviewer: Let me know if this is accurate!

Author: it's not, but I can update it

to serialize and deserialize dataclasses.

:::{important}

> If you're using Flytekit version below v1.11.1, your dataclass will need to decorate with `@dataclass_json` or

Reviewer: Just to clarify, this means that if you are using Flytekit < 1.11.1, you either need to decorate your dataclass with `@dataclass_json` or create a class that inherits `DataClassJSONMixin`, but that you don't need to do both? If you inherit from `DataClassJSONMixin`, do you need to do `from mashumaro.mixins.json import DataClassJSONMixin` as well?

Author: yes!

inherit from Mashumaro's `DataClassJSONMixin`.

> If you're using Flytekit version below v1.10, you'll need to decorate with `@dataclass_json` using

Reviewer: ```suggestion
If you're using Flytekit version less than v1.10, you'll need to add `from dataclasses_json import dataclass_json` to your imports  and decorate your dataclass with `@dataclass_json`.
```

> `from dataclasses_json import dataclass_json`

Reviewer: ```suggestion
```

to serialize and deserialize dataclasses.

:::{important}

> If you're using Flytekit version below v1.11.1, your dataclass will need to add `from dataclasses_json import dataclass_json` and decorate with `@dataclass_json`.

Reviewer: ```suggestion
If you're using Flytekit version below v1.11.1, you will need to add `from dataclasses_json import dataclass_json` to your imports and decorate your dataclass with `@dataclass_json`.
```

To begin, import the necessary dependencies:

```{rli} https://raw.githubusercontent.com/flyteorg/flytesnacks/cfb5ea3b0d0502ef7df1f2e14f4a0d9b78250b6a/examples/data_types_and_io/data_types_and_io/dataclass.py
:caption: data_types_and_io/dataclass.py
:lines: 1-9
```

> Build your custom image by ImageSpec:

Reviewer: ```suggestion
Build your custom image with ImageSpec:
```
